dart/lib/src/client.dart

```diff
@@ -77,6 +70,16 @@ abstract class SentryClient {
     return _options.transport.send(event);
   }
 
+  SentryEvent _prepareEvent(SentryEvent event) {
+    return event.copyWith(
+      serverName: _options.serverName,
+      dist: _options.dist,
+      environment: _options.environment,
+      release: _options.release,
+      platform: event.platform ?? sdkPlatform,
+    );
```

Comment:
I'm not sure if this works properly, does this work in the same way of:

`if event. serverName == null ? _options.serverName : event. serverName`?

Reply:
no the `copyWith` implementation is `options.serverName != null ? options.serverName : event.serverName`
but we could change its implementation to "fill" only the event's empty properties ?

Reply:
or here `serverName: event.serverName ?? _options.serverName,`

Reply:
we should only set a property to the event, read from elsewhere, if the event field is null, that means:

event. serverName == null, it should take the _options.serverName
if event. serverName != null, it should keep event. serverName

Reply:
✅